File: IUH_Olympic_ICPC/PhongTrao/bs_twopointer/N.py
def tinh(n, x, y):
    a=n*x
    b=n/2*y+n%2*x
    return min(a, b)

for _ in range(int(input())):
    n, m, x, y=map(int, input().split(" "))
    ans=0
    for i in range(n):
        s=input()
        s=s+'*'
        cnt=0
        for c in s: 
            if c=='.': cnt=cnt+1
            else: ans=ans+tinh(cnt); cnt=0
        print(ans)

            

File: IUH_Olympic_ICPC/LuyenACM_Hue/N.cpp
/*=====================================================================================
                Nothing is impossible, only you think it is impossible                 
                        Try, try, try again until you succeed     
                           Pratice, practice, and practice
I hated every minute of training, but I said, ‘Don’t quit. Suffer now and live the rest of your life as a champion.' - Mohamed Ali 
=====================================================================================*/
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll phi(ll x)
{
    if (x<=5) return 100;
    else return 100+(x-5);
}

void solve()
{
    ll n; cin>>n;
    map<string, ll> cnt; 
    for (ll i=0; i<n; i++)
    {
        string s; cin>>s;
        ++cnt[s];
    }
    ll ans=0;
    for (auto it: cnt) ans+=phi(it.second);
    cout<<ans;
}

int main()
{
    ios_base::sync_with_stdio(false); cin.tie(NULL);
    #ifndef ONLINE_JUDGE
    freopen("_input.txt", "r", stdin);
    freopen("_output.txt", "w", stdout);
    #endif
    solve();
    return 0;
}